### ISSUE TITLE
[$250] Copilot - App crashes after switching account

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#49939]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.41-2
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Email or phone of affected tester (no customers):** applausetester+kaaaa@applause.expensifail.com
**Issue reported by:** Applause - Internal Team

## Action Performed:
Precondition:
- - Log in as a copilot
1. Go to staging.new.expensify.com
2. Go to Inbox (important)
3. Go to Account settings
3. Click on the account switcher
4. Select another account

## Expected Result:
App will not crash

## Actual Result:
App crashes after switching account

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/user-attachments/assets/10405886-b99a-4aa1-b6ca-f85472774ab8

[logs (2).txt](https://github.com/user-attachments/files/17191972/logs.2.txt)

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021840797218276068437</li>
        <li>Upwork Job ID: 1840797218276068437</li>
        <li>Last Price Increase: 2024-09-30</li>
<li>Automatic offers: </li>
<ul>
<li>rojiphil | Reviewer | 104271871</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @rojiphil</details>

**Comment by bernhardoj** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
App crashes when switching account.

### What is the root cause of that problem?
As shown by the error on the console, the error caused by the `useOnyx` keys changes from a member collection key to the whole collection key (in this case, reportNameValuePairs_xx to reportNameValuePairs_).

The error is thrown specifically from https://github.com/Expensify/App/blob/36032e0427ea81802d63aa16eeef8948348fb11d/src/pages/home/report/ReportActionItem.tsx#L181.

When switching accounts, the report becomes undefined and we default the reportID attribute to an empty string.
https://github.com/Expensify/App/blob/36032e0427ea81802d63aa16eeef8948348fb11d/src/pages/home/ReportScreen.tsx#L169-L173

Because empty string is not undefined/null, `report?.reportID ?? -1` will still be an empty string.

### What changes do you think we should make in order to solve the problem?

We need to use || instead of ??
```
const [reportNameValuePairs] = useOnyx(`${ONYXKEYS.COLLECTION.REPORT_NAME_VALUE_PAIRS}${report?.reportID || '-1'}`);
```

I see that we use ?? for REPORT_NAME_VALUE_PAIRS in several places too, we can replace them all with || too.

**Comment by rojiphil** [comment]
Thanks for the proposals. 
The root cause is that the report id becomes an empty string. Falling back to `-1` will solve this issue.
@bernhardoj [proposal](https://github.com/Expensify/App/issues/49939#issuecomment-2383565131) LGTM
🎀👀🎀 C+ reviewed

**Comment by bernhardoj** [comment]
This is fixed in https://github.com/Expensify/App/pull/49255.

**Comment by rojiphil** [comment]
> This is fixed in #49255.

Ok. Let's close this then and move on cc @twisterdotcom @jasperhuangg
